libs/keycodes_ascii/src/lib.rs

```diff
@@ -241,138 +242,35 @@ pub enum Keycode {
-            40 => Some(Keycode::Quote),
-            41 => Some(Keycode::Backtick),
-            42 => Some(Keycode::LeftShift),
-            43 => Some(Keycode::Backslash),
-            44 => Some(Keycode::Z),
-            45 => Some(Keycode::X),
-            46 => Some(Keycode::C),
-            47 => Some(Keycode::V),
-            48 => Some(Keycode::B),
-            49 => Some(Keycode::N),
-            50 => Some(Keycode::M),
-            51 => Some(Keycode::Comma),
-            52 => Some(Keycode::Period),
-            53 => Some(Keycode::Slash),
-            54 => Some(Keycode::RightShift),
-            55 => Some(Keycode::PadMultiply), // Also PrintScreen
-            56 => Some(Keycode::Alt),
-            57 => Some(Keycode::Space),
-            58 => Some(Keycode::CapsLock),
-            59 => Some(Keycode::F1),
-            60 => Some(Keycode::F2),
-            61 => Some(Keycode::F3),
-            62 => Some(Keycode::F4),
-            63 => Some(Keycode::F5),
-            64 => Some(Keycode::F6),
-            65 => Some(Keycode::F7),
-            66 => Some(Keycode::F8),
-            67 => Some(Keycode::F9),
-            68 => Some(Keycode::F10),
-            69 => Some(Keycode::NumLock),
-            70 => Some(Keycode::ScrollLock),
-            71 => Some(Keycode::Home), // Also Pad7
-            72 => Some(Keycode::Up), // Also Pad8
-            73 => Some(Keycode::PageUp), // Also Pad9
-            74 => Some(Keycode::PadMinus),
-            75 => Some(Keycode::Left), // Also Pad4
-            76 => Some(Keycode::Pad5),
-            77 => Some(Keycode::Right), // Also Pad6
-            78 => Some(Keycode::PadPlus),
-            79 => Some(Keycode::End), // Also Pad1
-            80 => Some(Keycode::Down), // Also Pad2
-            81 => Some(Keycode::PageDown), // Also Pad3
-            82 => Some(Keycode::Insert), // Also Pad0
-            83 => Some(Keycode::Delete), // Also PadDecimal
-            84 => Some(Keycode::Unknown1),
-            85 => Some(Keycode::Unknown2),
-            86 => Some(Keycode::NonUsBackslash),
-            87 => Some(Keycode::F11),
-            88 => Some(Keycode::F12),
-            89 => Some(Keycode::Pause),
-            90 => Some(Keycode::Unknown3),
-            91 => Some(Keycode::SuperKeyLeft),
-            92 => Some(Keycode::SuperKeyRight),
-            93 => Some(Keycode::Menu),
-
-            _ => None,
-        }
-    }
-
-
-
     /// Obtains the ascii value for a keycode under the given modifiers
     pub fn to_ascii(&self, modifiers: KeyboardModifiers) -> Option<char> {
         // handle shift key being pressed
         if modifiers.is_shift() {
-            // if shift is pressed and caps lock is on, give a regular lowercase letter
             if modifiers.is_caps_lock() && self.is_letter() {
+            // if shift is pressed and caps lock is on, give a regular lowercase letter
                 return self.as_ascii();
-            }
+            } else {
```

Comment:
no need to reorder these. It's more readable (imo) to place the commend above the condition that it applies to, just like how we place doc comments right above the item they apply to.

Reply:
the `} else {` formatting OCD was too strong

Reply:
either one is ok in this scenario cuz the basic blocks within the conditional branches are very short. If you keep it as is, just fix the indentation to line up with the statement within the braces. Your call.